Make generate create two files without yaml extension in json format

## steps
- git clone current project
- make generate

## Result:
- two additional files without .yaml extension are created in json format:
prometheus-operator-0servicemonitorCustomResourceDefinition
prometheus-operator-0prometheusruleCustomResourceDefinition

![Snip20200429_3](https://user-images.githubusercontent.com/18525465/80573992-953d2900-8a33-11ea-9887-34b79f55a758.png)


Attach these two files
[manifests.zip](https://github.com/coreos/kube-prometheus/files/4550560/manifests.zip)

@dgrisonnet Thanks for the PR, I've tested on my mac and I got errors like below and the files without extension are not deleted:

```
lute import still works
./build.sh examples/kustomize.jsonnet
+ set -o pipefail
++ pwd
+ PATH=/Users/ben/src/qingcloud/kubesphere/kube-prometheus/tmp/bin:/Users/ben/go/bin:/Users/ben/go/bin:/Users/ben/go/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/usr/local/MacGPG2/bin:/usr/local/kubebuilder/bin:/usr/local/kubebuilder/bin
+ rm -rf manifests
+ mkdir -p manifests/setup
+ jsonnet -J vendor -m manifests examples/kustomize.jsonnet
+ xargs '-I{}' sh -c 'cat {} | gojsontoyaml > {}.yaml; find {} -delete' -- '{}'
find: -delete: manifests/../kustomization: relative path potentially not safe
find: -delete: manifests/alertmanager-alertmanager: relative path potentially not safe
find: -delete: manifests/alertmanager-secret: relative path potentially not safe
find: -delete: manifests/alertmanager-service: relative path potentially not safe
find: -delete: manifests/alertmanager-serviceAccount: relative path potentially not safe
find: -delete: manifests/alertmanager-serviceMonitor: relative path potentially not safe
find: -delete: manifests/grafana-dashboardDatasources: relative path potentially not safe
find: -delete: manifests/grafana-dashboardDefinitions: relative path potentially not safe
find: -delete: manifests/grafana-dashboardSources: relative path potentially not safe
find: -delete: manifests/grafana-deployment: relative path potentially not safe
find: -delete: manifests/grafana-service: relative path potentially not safe
find: -delete: manifests/grafana-serviceAccount: relative path potentially not safe
find: -delete: manifests/grafana-serviceMonitor: relative path potentially not safe
find: -delete: manifests/kube-state-metrics-clusterRole: relative path potentially not safe
find: -delete: manifests/kube-state-metrics-clusterRoleBinding: relative path potentially not safe
find: -delete: manifests/kube-state-metrics-deployment: relative path potentially not safe
find: -delete: manifests/kube-state-metrics-service: relative path potentially not safe
find: -delete: manifests/kube-state-metrics-serviceAccount: relative path potentially not safe
find: -delete: manifests/kube-state-metrics-serviceMonitor: relative path potentially not safe
find: -delete: manifests/node-exporter-clusterRole: relative path potentially not safe
find: -delete: manifests/node-exporter-clusterRoleBinding: relative path potentially not safe
find: -delete: manifests/node-exporter-daemonset: relative path potentially not safe
find: -delete: manifests/node-exporter-service: relative path potentially not safe
find: -delete: manifests/node-exporter-serviceAccount: relative path potentially not safe
find: -delete: manifests/node-exporter-serviceMonitor: relative path potentially not safe
find: -delete: manifests/prometheus-adapter-apiService: relative path potentially not safe
find: -delete: manifests/prometheus-adapter-clusterRole: relative path potentially not safe
find: -delete: manifests/prometheus-adapter-clusterRoleAggregatedMetricsReader: relative path potentially not safe
find: -delete: manifests/prometheus-adapter-clusterRoleBinding: relative path potentially not safe
find: -delete: manifests/prometheus-adapter-clusterRoleBindingDelegator: relative path potentially not safe
find: -delete: manifests/prometheus-adapter-clusterRoleServerResources: relative path potentially not safe
find: -delete: manifests/prometheus-adapter-configMap: relative path potentially not safe
find: -delete: manifests/prometheus-adapter-customMetricsApiService: relative path potentially not safe
find: -delete: manifests/prometheus-adapter-customMetricsClusterRoleBindingHPA: relative path potentially not safe
find: {}: No such file or directory
find: -delete: manifests/prometheus-adapter-customMetricsClusterRoleServerResources: relative path potentially not safe
find: -delete: manifests/prometheus-adapter-deployment: relative path potentially not safe
find: -delete: manifests/prometheus-adapter-roleBindingAuthReader: relative path potentially not safe
find: -delete: manifests/prometheus-adapter-service: relative path potentially not safe
find: -delete: manifests/prometheus-adapter-serviceAccount: relative path potentially not safe
find: -delete: manifests/prometheus-clusterRole: relative path potentially not safe
find: -delete: manifests/prometheus-clusterRoleBinding: relative path potentially not safe
find: -delete: manifests/prometheus-endpointsEtcd: relative path potentially not safe
find: -delete: manifests/prometheus-operator-serviceMonitor: relative path potentially not safe
find: -delete: manifests/prometheus-prometheus: relative path potentially not safe
find: -delete: manifests/prometheus-roleBindingConfig: relative path potentially not safe
find: -delete: manifests/prometheus-roleBindingSpecificNamespaces: relative path potentially not safe
find: -delete: manifests/prometheus-roleConfig: relative path potentially not safe
find: -delete: manifests/prometheus-roleSpecificNamespaces: relative path potentially not safe
find: -delete: manifests/prometheus-rules: relative path potentially not safe
find: -delete: manifests/prometheus-rulesEtcd: relative path potentially not safe
find: -delete: manifests/prometheus-secretEtcdCerts: relative path potentially not safe
find: -delete: manifests/prometheus-service: relative path potentially not safe
find: -delete: manifests/prometheus-serviceAccount: relative path potentially not safe
find: -delete: manifests/prometheus-serviceEtcd: relative path potentially not safe
find: -delete: manifests/prometheus-serviceKubeControllerManager: relative path potentially not safe
find: -delete: manifests/prometheus-serviceKubeScheduler: relative path potentially not safe
find: -delete: manifests/prometheus-serviceMonitor: relative path potentially not safe
find: -delete: manifests/prometheus-serviceMonitorApiserver: relative path potentially not safe
find: -delete: manifests/prometheus-serviceMonitorCoreDNS: relative path potentially not safe
find: -delete: manifests/prometheus-serviceMonitorEtcd: relative path potentially not safe
find: -delete: manifests/prometheus-serviceMonitorKubeControllerManager: relative path potentially not safe
find: -delete: manifests/prometheus-serviceMonitorKubeScheduler: relative path potentially not safe
find: -delete: manifests/prometheus-serviceMonitorKubelet: relative path potentially not safe
find: -delete: manifests/prometheus-serviceMonitorS2IOperator: relative path potentially not safe
find: -delete: manifests/setup/0namespace-namespace: relative path potentially not safe
find: {}: No such file or directory
find: {}: No such file or directory
find: {}: No such file or directory
find: {}: No such file or directory
find: {}: No such file or directory
find: {}: No such file or directory
find: -delete: manifests/setup/prometheus-operator-clusterRole: relative path potentially not safe
find: -delete: manifests/setup/prometheus-operator-clusterRoleBinding: relative path potentially not safe
find: -delete: manifests/setup/prometheus-operator-deployment: relative path potentially not safe
find: -delete: manifests/setup/prometheus-operator-service: relative path potentially not safe
find: -delete: manifests/setup/prometheus-operator-serviceAccount: relative path potentially not safe
make: *** [manifests] Error 1
```

--- a/README.md
+++ b/README.md
@@ -243,8 +243,12 @@ PATH="$(pwd)/tmp/bin:${PATH}"
 rm -rf manifests
 mkdir -p manifests/setup
 
-                                               # optional, but we would like to generate yaml, not json
-jsonnet -J vendor -m manifests "${1-example.jsonnet}" | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml; rm -f {}' -- {}
+# Calling gojsontoyaml is optional, but we would like to generate yaml, not json
+jsonnet -J vendor -m manifests "${1-example.jsonnet}" | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml' -- {}
+
+# Make sure to remove json files
+find manifests -type f ! -name '*.yaml' -delete
+rm kustomization
 
 ```
 

--- a/build.sh
+++ b/build.sh
@@ -14,6 +14,10 @@ PATH="$(pwd)/tmp/bin:${PATH}"
 rm -rf manifests
 mkdir -p manifests/setup
 
-                                               # optional, but we would like to generate yaml, not json
-jsonnet -J vendor -m manifests "${1-example.jsonnet}" | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml; rm -f {}' -- {}
+# Calling gojsontoyaml is optional, but we would like to generate yaml, not json
+jsonnet -J vendor -m manifests "${1-example.jsonnet}" | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml' -- {}
+
+# Make sure to remove json files
+find manifests -type f ! -name '*.yaml' -delete
+rm kustomization
 

